app/components/post/post.js

```diff
@@ -112,24 +146,91 @@ export default class Post extends Component {
         return attachments;
     }
 
+    renderMessage = (style, messageStyle, replyBar = false) => {
+        const {formatMessage} = this.props.intl;
+        const {currentUserId, post, roles} = this.props;
+        const actions = [];
+
+        // we should check for the user roles and permissions
+        // actions.push({text: 'Flag', onPress: () => console.log('flag pressed')}); //eslint-disable-line no-console
+        // if (post.user_id === currentUserId) {
+        //     actions.push({text: 'Edit', onPress: () => console.log('edit pressed')}); //eslint-disable-line no-console
+        // }
+        if (post.user_id === currentUserId || isAdmin(roles)) {
+            actions.push({text: formatMessage({id: 'post_info.del', defaultMessage: 'Delete'}), onPress: () => this.handlePostDelete()});
+        }
+
+        let messageContainer;
+        let message;
+        if (post.state === Constants.POST_DELETED) {
+            message = (
+                <FormattedText
+                    style={messageStyle}
+                    id='post_body.deleted'
+                    defaultMessage='(message deleted)'
+                />
+            );
+        } else {
+            message = (
+                <Text style={messageStyle}>
+                    {post.message}
+                </Text>
+            );
+        }
+
+        if (this.props.post.message.length) {
```

Comment:
This will prevent the menu from showing up for posts containing only files which I don't think we want